### PR TITLE
get_style_nodes should be compatible with parent method.

### PR DESCRIPTION
**Pull request by torounit**
## What?

Fix the following errors and fix phpdoc comment for get_styles_for_block.
 
> Warning: Declaration of WP_Theme_JSON_6_1::get_style_nodes($theme_json) should be compatible with WP_Theme_JSON_5_9::get_style_nodes($theme_json, $selectors = Array) in /var/www/html/wp-content/plugins/gutenberg/lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php on line 124

## Testing Instructions
1. npm run wp-env start 
2. show  http://localhost:8888/

### CODE REVIEW

**Comment by ramonjd** [comment]
Thanks for adding this patch @torounit 

I just added some minor formatting changes to kick the CI tests off again 😄 

The changes in https://github.com/WordPress/gutenberg/commit/323174aad891ec5d08929d2f53e15001dc9edc30#diff-f28aa79997745a4a51f917012b45a809c9b45d55134b55379a718f5c751dddf4R160 removed the use of `$selectors` in `get_style_nodes()`, so now it's an unused parameter and will trigger the lint warning.

```bash
124 | WARNING | Unused function parameter $selectors.
```

Do we throw in an ignore comment to suppress the warning?

```php
protected static function get_style_nodes( $theme_json, $selectors = array() ) { //phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
...
```


That would be an interim measure only, and not ideal.

I wonder if we should rather wind back the optimization to dedupe the code in `get_block_nodes`? 

Basically we put up with duplicated code in `get_style_nodes()` to retain the same function signature.

The other, harder way, might be to create a new function and deprecate `get_style_nodes()`. That might involve changing more files wherever `get_style_nodes()` is called to remove the second parameter.

What do you think @scruffian ?

cc @adamziel @draganescu

**Comment by ramonjd** [comment]
Actually, passing an optional `$selectors` arg to `get_block_nodes` might do the trick. 😆

**Comment by ramonjd** [comment]
I threw up a PR over in https://github.com/WordPress/gutenberg/pull/41218 in order to add a check that avoids the PHPCodeSniffer error

```bash
Fatal error: Uncaught TypeError: vsprintf(): Argument #2 ($values) must be of type array, string given in /app/vendor/squizlabs/php_codesniffer/src/Files/File.php:1056
```

File: lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php [hunk 1]
```diff
@@ -118,10 +118,11 @@ protected static function get_blocks_metadata() {
 	 *
 	 * @since 5.8.0
 	 *
-	 * @param array $theme_json     The tree to extract style nodes from.
+	 * @param array $theme_json The tree to extract style nodes from.
+	 * @param array $selectors  List of selectors per block.
 	 * @return array
 	 */
-	protected static function get_style_nodes( $theme_json ) {
+	protected static function get_style_nodes( $theme_json, $selectors = array()  ) {
```
Comment (andrewserong) [inline lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php]:
```suggestion
	protected static function get_style_nodes( $theme_json, $selectors = array() ) {
```

**Comment by ramonjd** [comment]
> I threw up a PR over in https://github.com/WordPress/gutenberg/pull/41218 in order to add a check that avoids the PHPCodeSniffer error

Thanks for spotting that space in https://github.com/WordPress/gutenberg/pull/41217/commits/bebac4c9e494656cb99c03770f3387c86fe34530 @andrewserong 

🤞 this one passes. I'll close my other test PR.

**Review by ramonjd — APPROVED** [review]
LGTM!

I tested according to the instructions over in https://github.com/WordPress/gutenberg/pull/41160

Thanks for putting this PR together @torounit !

**Comment by draganescu** [comment]
Missed the ping but this is 👌🏻 For a moment I wondered if we want to remove `$selectors` when this is merged, or do we still want it in `get_style_nodes`? It appears `get_block_nodes` needs it? `get_stylesheet` passes `$blocks_metadata` but that only is required on `get_block_nodes` which could also get it from `static::get_blocks_metadata() ` so it seems useless?

**Comment by ramonjd** [comment]
> For a moment I wondered if we want to remove $selectors when this is merged, or do we still want it in get_style_nodes? It appears get_block_nodes needs it? get_stylesheet passes $blocks_metadata but that only is required on get_block_nodes which could also get it from static::get_blocks_metadata() so it seems useless


Thanks @draganescu 

You're right on both counts: we don't need `$selectors` in `get_style_nodes()` anymore. `static::get_blocks_metadata()` doesn't really need it either 🤣 

The intention behind this PR was to address the PHP warning

> Warning: Declaration of WP_Theme_JSON_6_1::get_style_nodes($theme_json) should be compatible with WP_Theme_JSON_5_9::get_style_nodes($theme_json, $selectors = Array) in /var/www/html/wp-content/plugins/gutenberg/lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php on line 124

So we added `$selectors = array()` to the `get_style_nodes()` function arguments to make it compatible with the 5.9 method.

Then the linter started complaining that `$selectors` was never used in `get_style_nodes()` 😄 

So passing `$selectors` down from `get_style_nodes()` to `static::get_blocks_metadata()` was a quick way to end the complaints and warnings.

What do you think a valid path to deprecate the argument might be?

I was thinking in a new PR we could:

1. revert this PR
2. remove  `get_style_nodes()` from WP_Theme_JSON_5_9
3. migrate any methods in WP_Theme_JSON_5_9 that use `get_style_nodes()` to WP_Theme_JSON_6_1 and remove the second argument
4. Make sure we port these changes to WordPress 6.1 when that comes around 😄 

Here's a PR that does all that:

- https://github.com/WordPress/gutenberg/pull/41262

**Comment by oandregal** [comment]
Hey, while working on https://github.com/WordPress/gutenberg/pull/46579 I realized this PR was not backported to WordPress 6.1. When we land that PR and then backport to WordPress 6.2 everything will be fine, but I wanted to raise in case you think this should be ported for a potential WordPress 6.1.X release.

**Comment by aaronrobertshaw** [comment]
It looks like the fact this PR wasn't backported is still causing some knock on effects.

For example, https://github.com/WordPress/gutenberg/pull/62465 needs to update the `get_block_nodes` signature but this PR modified it without backporting that change. When backporting #62465, I'll update the `get_block_nodes` function to match Gutenberg but it would be great if anyone that worked on this could double check those changes.
